importer/importer.go

```diff
@@ -21,68 +23,47 @@ func NewAccountingServiceImpl(vms chan<- *goat_grpc.VmRecord, ips chan<- *goat_g
 	}
 }
 
-// ProcessVms is a GRPC call -- do not use!
-func (asi AccountingServiceImpl) ProcessVms(stream goat_grpc.AccountingService_ProcessVmsServer) error {
-	for {
-		data, err := stream.Recv()
-
-		if err == io.EOF {
-			return stream.SendAndClose(&goat_grpc.Confirmation{
-				Accepted: true,
-				Msg:      "ok",
-			})
-		}
-
-		if err != nil {
-			return err
-		}
+// Process is a GRPC call -- do not use!
+func (asi AccountingServiceImpl) Process(stream goat_grpc.AccountingService_ProcessServer) error {
+	id, err := stream.Recv()
+	if err != nil {
+		return err
+	}
 
-		for _, vm := range data.Vms {
-			asi.vmConsumer <- vm
-		}
+	clientID := id.GetIdentifier()
```

Comment:
Is this really the way how to determine which one of that oneof field is used? It wouldn't surprise me since it's Go but still I'm better asking.

Reply:
I agree :+1:. That looks like a better solution. And I've seen it before so it's probably a common one in Go.